sdk/maps/maps-geolocation-rest/README.md

```diff
@@ -0,0 +1,138 @@
+provide an instance of the desired credential type obtained from the
+[@azure/identity](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) library.
+
+To authenticate with AAD, you must first `npm` install [`@azure/identity`](https://www.npmjs.com/package/@azure/identity)
+
+After setup, you can choose which type of [credential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) from `@azure/identity` to use.
+As an example, [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#defaultazurecredential)
+can be used to authenticate the client.
+
+You'll need to register the new Azure AD application and grant access to Azure Maps by assigning the required role to your service principal. For more information, see [Host a daemon on non-Azure resources](https://learn.microsoft.com/azure/azure-maps/how-to-secure-daemon-app#host-a-daemon-on-non-azure-resources). Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+`AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.
+
+You will also need to specify the Azure Maps resource you intend to use by specifying the `clientId` in the client options.
+The Azure Maps resource client id can be found in the Authentication sections in the Azure Maps resource. Please refer to the [documentation](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication#view-authentication-details) on how to find it.
+
+```javascript
+const MapsGeolocation = require("@azure-rest/maps-geolocation").default;
+const { DefaultAzureCredential } = require("@azure/identity");
+const credential = new DefaultAzureCredential();
+const client = MapsGeolocation(credential, "<maps-account-client-id>");
+```
+
+#### Using a Subscription Key Credential
+
+You can authenticate with your Azure Maps Subscription Key. Please install the `@azure/core-auth` package:
+
+```bash
+npm install @azure/core-auth
+```
+
+```javascript
+const MapsGeolocation = require("@azure-rest/maps-geolocation").default;
+const AzureKeyCredential = require("@azure/core-auth");
+const credential = new AzureKeyCredential("<subscription-key>");
+const client = MapsGeolocation(credential);
+```
+
+## Key concepts
+
+### MapsGeolocationClient
+
+`MapsGeoClient` is the primary interface for developers using the Azure Maps Geolocation client library. Explore the methods on this client object to understand the different features of the Azure Maps Geolocation service that you can access.
+
+## Examples
+
+You can get the country code from a IP address:
+
+```javascript
+const result = await client.path("/geolocation/ip/{format}", "json").get({
+  queryParameters: { ip: "2001:4898:80e8:b::189" },
+});
+
+if (isUnexpected(result)) {
```

Comment:
You doesn't import `isUnexpected` in the example

Reply:
Updated in [eb12dac](https://github.com/Azure/azure-sdk-for-js/pull/23956/commits/eb12dac47f44dbf4ffa3df8ab627bea3d4d2ce1d)